indykite/resource_external_data_resolver.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2024 IndyKite
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indykite
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/indykite/indykite-sdk-go/config"
+	configpb "github.com/indykite/indykite-sdk-go/gen/indykite/config/v1beta1"
+)
+
+const (
+	externalDataResolverURLKey              = "url"
+	externalDataResolverMethodKey           = "method"
+	externalDataResolverHeadersKey          = "headers"
+	externalDataResolverRequestTypeKey      = "request_type"
+	externalDataResolverRequestPayloadKey   = "request_payload"
+	externalDataResolverResponseTypeKey     = "response_type"
+	externalDataResolverResponseSelectorKey = "response_selector"
+)
+
+var (
+	externalDataResolverHeaderRegex = regexp.MustCompile(`^[a-zA-Z0-9-]+$`)
+)
+
+func resourceExternalDataResolver() *schema.Resource {
+	readContext := configReadContextFunc(resourceExternalDataResolverFlatten)
+
+	return &schema.Resource{
+		Description: "ExternalDataResolver is a configuration node that allows to fetch data from external sources",
```

Comment:
is it a `node`?

Reply:
No thank you, I am so old I still think config nodes are nodes

Reply:
unfortunately a lot of the public code still says "node" and we can't change that without making breaking changes..